sycl/doc/extensions/AsyncGroupCopy/async_group_copy.asciidoc

```diff
@@ -0,0 +1,115 @@
+# Async Group Copy Extension for DPC++: SYCL_EXT_ONEAPI_ASYNC_GROUP_COPY
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+:dpcpp: pass:[DPC++]
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+Copyright (c) 2021-2021 Intel Corporation.  All rights reserved.
+
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension is written against the SYCL 2020 revision 3 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+## Contributors
+
+* Gordon Brown
+* Tadej Ciglaric
+* Jack Kirk
+* Finlay Marno
+
+## Introduction
+
+This document proposes that `async_work_group_copy` be deprecated and replaced 
+with `async_group_copy`. `async_group_copy` is a non-member function, in line
+with the other group functions, which generalizes `async_work_group_copy` to
+also work with sub-groups.
```

Comment:
I expect there should be a restriction that all the work-items in the group would call the function with the same arguments. I don't think convergence is required. Could you suggest where it might fit better. 
I think this relates a bit to one of John's questions https://github.com/intel/llvm/pull/4950#discussion_r751403933

Reply:
I don't think this extension document needs to decide where in the SYCL 2020 spec these functions would eventually be described.  That will be decided later by the Khronos SYCL group if/when this extension is adopted into a future version of the core SYCL spec.

The extension spec just need to document the behavior of the new extended APIs you propose.  The current description is somewhat unclear because it says "in line with the other group functions", but it sounds like these APIs are not perfectly in line with them because they do not need to be in convergent code.  It would be better to just describe the requirements of the new APIs.  It sounds like it is a requirement that all work items in the group must call the API passing the same arguments.  You should say this somewhere in the extension spec.

Reply:
I think I had an incorrect understanding of converged control flow. I don't believe (currently) it should be a requirement that the work-items are synchronized, but all the work-items should call `async_group_copy` with the same argument while in converged control flow.

Reply:
I've added a bit saying `async_group_copy` requires convergent control flow and removed the line about it being 'in line with the other group functions'